### PR TITLE
Feature/incoming mail

### PR DESCRIPTION
**Pull request by meskio**

### CODE REVIEW

File: src/leap/bitmask/services/mail/imap.py [hunk 1]
```diff
@@ -70,3 +72,24 @@ def start_imap_service(*args, **kwargs):
...
+        incoming_mail = IncomingMail(
+            keymanager,
+            soledad,
+            inbox,
+            userid)
+        incoming_mail.start_loop()
+        return incoming_mail
+
+    # XXX: do I really need to know here how to get a mailbox??
+    d = imap_factory.getMailbox(INBOX_NAME)
```
Comment (kalikaneko) [inline src/leap/bitmask/services/mail/imap.py]: I think the best next step now is to make Mail a composite service, made out of ServiceIncoming, IMAPService and SMTPService, and hiding all initialization complexity from the eyes of the client backend.

File: src/leap/bitmask/services/mail/imapcontroller.py [hunk 2]
```diff
@@ -58,16 +58,21 @@ def start_imap_service(self, userid, offline=False):
...
             logger.debug("Starting loop")
-            self.imap_service.start_loop()
+            d = imap.start_incoming_mail_service(
+                self._keymanager,
+                self._soledad,
+                self.imap_factory,
+                userid)
+            d.addCallback(assign_incoming_mail)
 
     def stop_imap_service(self, cv):
```
Comment (kalikaneko) [inline src/leap/bitmask/services/mail/imapcontroller.py]: do you think we still need the condition here?

File: src/leap/bitmask/services/mail/imap.py (hunk 1 above)
Comment (meskio) [inline src/leap/bitmask/services/mail/imap.py]: Agree.

File: src/leap/bitmask/services/mail/imapcontroller.py (hunk 2 above)
Comment (meskio) [inline src/leap/bitmask/services/mail/imapcontroller.py]: I don't know where the offline flag comes from, is it still used, if it has any meaning I guess if we are offline we should not start incoming mail.

Comment (kalikaneko) [inline src/leap/bitmask/services/mail/imapcontroller.py]: I was referring to the threading.Condition
the offline flag is meant, mostly, to ease development in the client, bypassing the need for login and stablishing a session before unlocking the local soledad store.
it's broken right now and disabled until we're in good shape with 0.9

Comment (meskio) [inline src/leap/bitmask/services/mail/imapcontroller.py]: I didn't pay attention on what is this threading.Condition, and how it is used. But from the docstring it looks pretty much like a Deferred.

**Pull request merged**